### PR TITLE
Add initial CircleCI configuration

### PR DESCRIPTION
**Pull request by ijc**
Signed-off-by: Ian Campbell <ijc@docker.com>

### CODE REVIEW

**Comment by justincormack** [comment]
Would it make sense to build these in `alpine` not `stretch`?

**Comment by amirmc** [comment]
Possibly the wrong place to ask but are issues meant to be turned off in this repo?
If so, a note in the readme would be helpful :)

**Comment by ijc** [comment]
slack would have been better.

issues are off until we a ready to cut over.

**Comment by ijc** [comment]
@justincormack some of the circle stuff (such as the mechanisms for setting $PATH) rely on `/bin/sh` being `bash` not busybox, it could probably be made to work, but for now the path of least resistance is to stick with Debian instead of alpine.

**Comment by ijc** [comment]
OK, this adds a reasonable enough baseline (builds all the packages and checks the yml files are up to date). Please review so we can merge this and I will add image building and other improvements in one or more followups.
/cc @rn @justincormack

**Comment by justincormack** [comment]
Ok. The circleci config is looking a bit too much like a Makefile for my taste, can we move stuff into the actual Makefile?

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+          command: mkdir -p /workspace/bin
+      - run:
+          name: Install packages
+          command: apt-get update && apt-get install -y ca-certificates curl
+      - run:
+          name: Fetch binaries
+          command: |
+            curl -fsSL -o /workspace/bin/linuxkit https://152-46932243-gh.circle-artifacts.com/0/linuxkit-linux-amd64
+            curl -fsSL -o /workspace/bin/moby https://309-89286561-gh.circle-artifacts.com/0/moby-linux
+            curl -fsSL -o /tmp/docker.tgz https://download.docker.com/linux/static/stable/x86_64/docker-17.06.2-ce.tgz
```
Comment (rn) [inline .circleci/config.yml]: Is there a reason you are downloading `docker` here while above you only say:
```
- setup_remote_docker:
     version: 17.06.1-ce
```

Comment (ijc) [reply .circleci/config.yml]: Download is the client, the `setup_remote_docker` is the server. 17.06.1 is the latest point of 17.06 circle provides for the server side which is why the client is a bit newer.

**Pull request merged**